admin/app/helpers/solidus_admin/permission_sets_helper.rb

```diff
@@ -0,0 +1,109 @@
+    #   safest way to reliably match on these PermissionSets. Other
+    #   optimizations that were tested resulted in too many false matches on
+    #   other non-stock-Solidus but similarly named PermissionSets.
+    # @param permission_sets [Array<Spree::PermissionSet>] an array of
+    #   PermissionSet objects to be organized into categories based on their
+    #   names.
+    # @param view_string [String] A string of your choice associated with "View"
+    #   or "Display" level permissions. Used when rendering the checkbox.
+    # @param edit_string [String] A string of your choice associated with "Edit"
+    #   or "Management" level permissions. Used when rendering the checkbox.
+    def organize_permissions(permission_sets:, view_string:, edit_string:)
+      return {} if permission_sets.blank?
+
+      permission_sets.each_with_object({}) do |permission, grouped_permissions|
+        case permission.name
+        when /OrderDisplay/i
+          grouped_permissions[:order] ||= []
+          grouped_permissions[:order] << [view_string, permission.id]
+        when /OrderManagement/i
+          grouped_permissions[:order] ||= []
+          grouped_permissions[:order] << [edit_string, permission.id]
+        when /ConfigurationDisplay/i
+          grouped_permissions[:configuration] ||= []
+          grouped_permissions[:configuration] << [view_string, permission.id]
+        when /ConfigurationManagement/i
+          grouped_permissions[:configuration] ||= []
+          grouped_permissions[:configuration] << [edit_string, permission.id]
+        when /UserDisplay/i
+          grouped_permissions[:user] ||= []
+          grouped_permissions[:user] << [view_string, permission.id]
+        when /UserManagement/i
+          grouped_permissions[:user] ||= []
+          grouped_permissions[:user] << [edit_string, permission.id]
+        when /RestrictedStockDisplay/i
+          grouped_permissions[:restricted_stock] ||= []
+          grouped_permissions[:restricted_stock] << [view_string, permission.id]
+        when /RestrictedStockManagement/i
+          grouped_permissions[:restricted_stock] ||= []
+          grouped_permissions[:restricted_stock] << [edit_string, permission.id]
+        when /StockDisplay/i
+          grouped_permissions[:stock] ||= []
+          grouped_permissions[:stock] << [view_string, permission.id]
+        when /StockManagement/i
+          grouped_permissions[:stock] ||= []
+          grouped_permissions[:stock] << [edit_string, permission.id]
+        when /ProductDisplay/i
+          grouped_permissions[:product] ||= []
+          grouped_permissions[:product] << [view_string, permission.id]
+        when /ProductManagement/i
+          grouped_permissions[:product] ||= []
+          grouped_permissions[:product] << [edit_string, permission.id]
+        else
+          grouped_permissions[:other] ||= []
+          grouped_permissions[:other] << [permission.name, permission.id]
+        end
```

Comment:
I think handling the "other" permissions like this is a smart idea.

Reply:
With a `permission_type` column we could gather all custom (non-core) permission set classes under this group and maybe even log a deprecation warning so that stores know that they need to add the `self.permission_type` method to their classes.

Reply:
Great suggestions from you both, thanks for the reviews! One thing to note is that the default permissions that come with `core` already have 3 permissions that fall outside of the usual `view` / `edit` dichotomy, `SuperUser`, `DefaultCustomer`, and `DashboardDisplay`. So I've found that `other` is a useful tool for displaying those regardless of how we choose to re-group/re-identify our other permission sets. Still happy to add a deprecation warning perhaps when no `self.permission_type` is present though!

<img width="615" alt="Screenshot 2024-09-04 at 11 06 27 AM" src="https://github.com/user-attachments/assets/48c4a3dd-90ee-4485-842e-7ffb27886cab">

Reply:
This is the general gist of the changes I intent to make to the seeds file:

```
[{:name=>"UserManagement", :privilege=>"user", :category=>"management"},
 {:name=>"UserDisplay", :privilege=>"user", :category=>"display"},
 {:name=>"SuperUser", :privilege=>"super_user", :category=>"other"},
 {:name=>"StockManagement", :privilege=>"stock", :category=>"management"},
 {:name=>"StockDisplay", :privilege=>"stock", :category=>"display"},
 {:name=>"RestrictedStockManagement", :privilege=>"restricted_stock", :category=>"management"},
 {:name=>"RestrictedStockDisplay", :privilege=>"restricted_stock", :category=>"display"},
 {:name=>"ProductManagement", :privilege=>"product", :category=>"management"},
 {:name=>"ProductDisplay", :privilege=>"product", :category=>"display"},
 {:name=>"OrderManagement", :privilege=>"order", :category=>"management"},
 {:name=>"OrderDisplay", :privilege=>"order", :category=>"display"},
 {:name=>"DefaultCustomer", :privilege=>"default_customer", :category=>"other"},
 {:name=>"DashboardDisplay", :privilege=>"dashboard", :category=>"other"},
 {:name=>"ConfigurationManagement", :privilege=>"configuration", :category=>"management"},
 {:name=>"ConfigurationDisplay", :privilege=>"configuration", :category=>"display"}]

Reply:
Updates are up, resolving this for now, always happy to re-open for more suggestions!